### PR TITLE
Fix generation for services that contain operations with the same name as the service

### PR DESCRIPTION
**Pull request by adamthom-amzn**
## Description
If a 'FooService' contains an operation named 'Foo', the generator will attempt to generate a 'FooRequest' that extends an abstract 'FooRequest'.  This adds 'Default' to the concrete class names only in cases where there is an operation with a clashing name.

## Motivation and Context
Some simpler microservices do not adhere to the standard naming guidelines.  This allows clients to be generated for them.

## Testing
This should only impact models for which non-compiling code would be generated today.  This was tested with a simple unit test, and out-of-band against a similar service model end-to-end.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by millems** [comment]
Wouldn't this change make it backwards-incompatible to add an operation with the same name as the service, since it would rename an existing class?

Would it be better to rename the operation?

**Comment by adamthom-amzn** [comment]
> Wouldn't this change make it backwards-incompatible to add an operation with the same name as the service, since it would rename an existing class?
> 
> Would it be better to rename the operation?

Yeah, though I think it's unlikely someone would do that, it's a fair point.  I will see if I can figure out what a decent heuristic for renaming the operation would be.

**Comment by adamthom-amzn** [comment]
Changed it to change the generation of request/response/exception implementations rather than the base class.  As discussed offline, I think this is a bit more understandable than changing the name of the operation (as I think customers will expect the operation name to align with the model and its documentation more than the request type name), though I can also explore renaming the operation if the naming convention change for the request types seems too invasive upon review.

I also don't know why AddEmptyInputShape/AddEmptyOutputShape were not delegating naming of the shapes to the naming strategy; if that was not an oversight, I can revisit those changes as well.

File: codegen/src/test/java/software/amazon/awssdk/codegen/IntermediateModelBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                                            .getResource("poet/client/c2j/collision/service-2.json").getFile());
+        IntermediateModel testModel = new IntermediateModelBuilder(
+            C2jModels.builder()
+                     .serviceModel(ModelLoaderUtils.loadModel(ServiceModel.class, modelFile))
+                     .customizationConfig(CustomizationConfig.create())
+                     .build())
+            .build();
+
+        assertEquals(new HashSet<>(Arrays.asList("DefaultCollisionException", "DefaultCollisionRequest", "DefaultCollisionResponse")),
+                     testModel.getShapes().values().stream().map(s -> s.getShapeName()).collect(Collectors.toSet()));
```
Comment (millems) [inline codegen/src/test/java/software/amazon/awssdk/codegen/IntermediateModelBuilderTest.java:44]:
```java
assertThat(testModel.getShapes().values())
    .extracting(ShapeModel::getShapeName)
    .containsExactlyInAnyOrder("DefaultCollisionException", "DefaultCollisionRequest", "DefaultCollisionResponse");
```
?

Comment (adamthom-amzn) [reply codegen/src/test/java/software/amazon/awssdk/codegen/IntermediateModelBuilderTest.java]: Changed.

**Pull request merged**